domain/src/main/java/com/teamagam/gimelgimel/domain/map/entities/symbols/Symbol.java

```diff
@@ -2,6 +2,7 @@
 
 import com.teamagam.gimelgimel.domain.map.entities.interfaces.ISymbolVisitable;
 
-public interface Symbol extends ISymbolVisitable{
+public interface Symbol extends ISymbolVisitable {
 
+    boolean isSelected();
```

Comment:
I'm not sure that every symbol should have the "selectable" behavior. So maybe don't force it in the interface but as I suggested, convert the BaseSymbol class to SelectableSymbol class...